### PR TITLE
Refactor: change the order of #inlcude in some functions

### LINKED ISSUE #3053: The order of header files is not following the standard.
### Describe the Code Quality Issue

In most cpp files, the order of header files is not following the [c++ style](https://mcresearch.gitee.io/abacus-user-guide/develop-C++.html#3-include-%E9%A1%BA%E5%BA%8F%EF%BC%88clang-format-%E5%8F%AF%E4%BB%A5%E8%87%AA%E5%8A%A8%EF%BC%89)

### Additional Context

_No response_

### Task list for Issue attackers

- [ ] Identify the specific code file or section with the code quality issue.
- [ ] Investigate the issue and determine the root cause.
- [ ] Research best practices and potential solutions for the identified issue.
- [ ] Refactor the code to improve code quality, following the suggested solution.
- [ ] Ensure the refactored code adheres to the project's coding standards.
- [ ] Test the refactored code to ensure it functions as expected.
- [ ] Update any relevant documentation, if necessary.
- [ ] Submit a pull request with the refactored code and a description of the changes made.

### PR DESCRIPTION
**Pull request by Satinelamp**
fix #3053 by changing the order of `#include` to meet the requirement of [c++ style](https://mcresearch.gitee.io/abacus-user-guide/develop-C++.html#3-include-%E9%A1%BA%E5%BA%8F%EF%BC%88clang-format-%E5%8F%AF%E4%BB%A5%E8%87%AA%E5%8A%A8%EF%BC%89)

some small fixes in the doc.

### CODE REVIEW

**Comment by caic99** [comment]
@Satinelamp What's the point of keeping `#include <ctime>`?

**Comment by Satinelamp** [comment]
@caic99 
I did some tests, it turns out we could remove `<ctime>` in the main function, I guess this is a legacy code.

**Comment by dyzheng** [comment]
I have no problem of the Refactor part for this PR, but the Feature part not related to the Issue.

**Comment by dyzheng** [comment]
Could you add a Issue for this PR?

**Comment by Satinelamp** [comment]
@dyzheng, I just add an issue #3053

File: source/module_base/parallel_global.h [hunk 1]
```diff
@@ -11,46 +11,56 @@
...
+	 * @brief An interface function to call "Parallel_Global::divide_pools()" 
+	 * 
+	 */
 	void init_pools();
-	void divide_pools(void);
-}
+    
+    void divide_pools(void);
+	
+	/**
```
Comment (dyzheng) [inline source/module_base/parallel_global.h]: Please format your code, use space but not the tab for indentation.

File: source/driver.cpp [hunk 2]
```diff
@@ -62,7 +62,7 @@ void Driver::reading(void)
 
 #ifdef __MPI
     // (4)  divide the GlobalV::NPROC processors into GlobalV::KPAR for k-points parallelization.
-    Parallel_Global::init_pools();
+    Parallel_Global::divide_pools();
```
Comment (dyzheng) [inline source/driver.cpp]: The call of init_pools() change to divide_pools(), which looks not related to the Issue or the title of PR?

**Review by dyzheng — APPROVED** [review]
